### PR TITLE
Split dual functionality of CurrentPlaylistItem

### PR DESCRIPTION
**Pull request by smoogipoo**
In trying to facilitate https://github.com/ppy/osu/pull/17657, I found many weird usages of `CurrentPlaylistItem`, derived from the fact that it doesn't update in the case that the room is created but the composite is not in the match sub screen (as per `TestSceneMatchStartControl`).

I've removed this dual usage of `CurrentPlaylistItem` as a mechanism for tracking both the room's current state and the user's selection (inside a Playlist room). It now always tracks the room's current state. A bindable is passed down for cases where the subscreen's selection is necessary instead.

### CODE REVIEW

File: osu.Game.Tests/Visual/Multiplayer/TestSceneMatchStartControl.cs [hunk 1]
```diff
@@ -68,10 +66,19 @@ private void load(GameHost host, AudioManager audio)
...
                 }
             };
         });
 
+        protected override void Update()
+        {
+            base.Update();
+
+            if (RoomJoined)
+                selectedItem.Value = MultiplayerClient.APIRoom?.Playlist.SingleOrDefault(i => i.ID == MultiplayerClient.Room?.Settings.PlaylistItemId);
```
Comment (peppy) [inline osu.Game.Tests/Visual/Multiplayer/TestSceneMatchStartControl.cs:79]: This is a bit scary. Might warrant an inline comment explaining why this is needed (ie. what piece of the flow is missing in this test setup which would usually handle this).

File: osu.Game.Tests/Visual/Multiplayer/TestSceneMultiplayerSpectateButton.cs [hunk 2]
```diff
@@ -77,12 +75,21 @@ private void load(GameHost host, AudioManager audio)
...
                     }
                 }
             };
         });
 
+        protected override void Update()
+        {
+            base.Update();
+
+            if (RoomJoined)
```
Comment (peppy) [inline osu.Game.Tests/Visual/Multiplayer/TestSceneMultiplayerSpectateButton.cs:89]: As above, I'd prefer there be some kind of explanation.

File: osu.Game/Screens/OnlinePlay/Multiplayer/Match/MatchStartControl.cs [hunk 3]
```diff
@@ -14,12 +14,15 @@
...
 using osu.Game.Online.Multiplayer;
 using osu.Game.Online.Multiplayer.Countdown;
+using osu.Game.Online.Rooms;
 using osuTK;
 
 namespace osu.Game.Screens.OnlinePlay.Multiplayer.Match
 {
     public class MatchStartControl : MultiplayerRoomComposite
     {
+        public readonly IBindable<PlaylistItem> SelectedItem = new Bindable<PlaylistItem>();
```
Comment (peppy) [inline osu.Game/Screens/OnlinePlay/Multiplayer/Match/MatchStartControl.cs:24]: Please add xmldoc. I'm still confused as to the difference between `Current` and `Selected` items.

**Comment by peppy** [comment]
On an initial read I'm having trouble grasping the difference between the `Current` and `Selected` terminology. I wonder if we can come up with better names for one of them?

**Comment by smoogipoo** [comment]
> On an initial read I'm having trouble grasping the difference between the `Current` and `Selected` terminology

I'm honestly not surprised, and had a feeling this would come up.

**Comment by bdach** [comment]
Tests look rather red here.

**Comment by smoogipoo** [comment]
I can't resolve the test failures. They're related to the `TestMultiplayerClient` re-using the client-side `MultiplayerRoom`.

Just going to close this PR for now.

**Pull request closed without merging**